### PR TITLE
CEPH-83573327 - Verify image migration from Ec pool to Ec with default k_m values

### PR DESCRIPTION
**Pull request by rkhambad**
Ticket :  https://issues.redhat.com/browse/RHCEPHQE-7871
This PR includes Automation of image migration test scenario
verify Migration of the images source EC pools to destination EC pool with default k_m values.
Polarion Link : https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83573327

File changed :
[suites/pacific/rbd/tier-2_rbd_regression.yaml] - Added test suite
[suites/quincy/rbd/tier-2_rbd_regression.yaml] - Added test suite

Module file Renamed and changed : 
older file :  [tests/rbd/rbd_image_Live_migration_rep2Rep.py]
new renamed file : [tests/rbd/rbd_image_migration.py]

success log :
5.3 : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4z1go/
6.0 : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-frchs/

### CODE REVIEW

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        # commit migration process
+        rbd.migration_action(action="commit", dest_spec=dest_spec)
+
+        # verify commit migration
+        verify_migration_commit(rbd, ecpool2, image)
+
+        # Display image info and verify
+        image_info = rbd.image_info(ecpool2, image)
+        log.info(f"Image info : {image_info}")
+        if image not in image_info["name"]:
```
Comment (sunilangadi2) [inline tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:98]: Irrespective of commit migration action specified `image` name will be in image_info Is it?

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            )
+            return 1
+        log.info(
+            "Migration is successfully completed from EC pool to EC pool for default K_m values"
+        )
+        return 0
+
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
```
Comment (sunilangadi2) [inline tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:110]: please add pool cleanup in finally clause?

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py (hunk 1 above)
Comment (rkhambad) [reply tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:98]: discuss in review meeting. closing as per discussion.

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        # commit migration process
+        rbd.migration_action(action="commit", dest_spec=dest_spec)
+
+        # verify commit migration
+        verify_migration_commit(rbd, ecpool2, image)
+
+        # Display image info and verify
+        image_info = rbd.image_info(ecpool2, image)
+        log.info(f"Image info : {image_info}")
```
Comment (VasishtaShastry) [inline tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:97]: [Optional]
I think image info can be pushed to debug

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        if image not in image_info["name"]:
+            log.error(
+                f"Image {image} is not listed in 'rbd image info' for pool {ecpool2}"
```
Comment (VasishtaShastry) [inline tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:100]: `rbd image info` is not a list right ?
I think we need to make this message more meaningful/apt

**Review by VasishtaShastry — DISMISSED** [review]

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        "ec-pool-k-m": "2,2",
+        "do_not_create_image": True,
+        "crush-failure-domain": "osd",
+        "ec_pool_config": {
+            "pool": "rbd_ECpool2",
+            "data_pool": "data_pool_ec2",
+        },
+    }
+    kw["config"].update(config)
+    log.info(kw["config"])
```
[the remark this answers is not in the document]
Comment (rkhambad) [reply tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:68]: Done.

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py (hunk 3 above)
Comment (rkhambad) [reply tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:97]: Done.

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py (hunk 4 above)
Comment (rkhambad) [reply tests/rbd/rbd_migration_ec2ec_dafault_k_m.py]: done. Please check if it is fine now.

File: tests/rbd/rbd_migration_ec2ec_dafault_k_m.py (hunk 2 above)
Comment (rkhambad) [reply tests/rbd/rbd_migration_ec2ec_dafault_k_m.py:108]: Done.

**Review by sunilangadi2 — APPROVED** [review]
changes LGTM

**Review by sunilkumarn417 — CHANGES_REQUESTED** [review]
please address the changes as mentioned in the PR-2275

**Comment by rkhambad** [comment]
> please address the changes as mentioned in the PR-2275

Done. Addressed all new changes.

File: suites/pacific/rbd/tier-2_rbd_regression.yaml [hunk 6]
```diff
@@ -168,12 +168,6 @@ tests:
       name: Test image migration on ecpool with different K_m values
       polarion-id: CEPH-83573322
 
-  - test:
```
Comment (sunilangadi2) [inline suites/pacific/rbd/tier-2_rbd_regression.yaml:171]: why this test got deleted?

Comment (rkhambad) [reply suites/pacific/rbd/tier-2_rbd_regression.yaml:171]: This been deleted and added at line #184
In last PR we are using different module file for rep pool to rep pool image migration, now i changed some code and make single module which will take care of rep pool to rep pool as well as ec pool to ec pool migration.

**Review by rahullepakshi — APPROVED** [review]

**Comment by VasishtaShastry** [comment]
lgtm from the perspective of comments that were given by me.

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**